marquez_airflow/extractors/bigquery_extractor.py

```diff
@@ -72,16 +62,45 @@ def _bq_table_name(self, bq_table):
     # convenience method
     def extract_on_complete(self, task_instance) -> [StepMetadata]:
         log.debug(f"extract_on_complete({task_instance})")
-
-        source = self._source()
-        bigquery_job_id = self._get_bigquery_job_id(task_instance)
-
         inputs = None
         outputs = None
         context = {
             'sql': self.operator.sql,
-            'bigquery.job_id': bigquery_job_id
         }
+        try:
+            source = self._source()
+            bigquery_job_id = self._get_bigquery_job_id(task_instance)
+        except Exception as e:
+            log.error(f"Cannot retrieve job details from BigQuery.Client. {e}",
+                      exc_info=True)
+            try:
+                sql_meta = SqlParser.parse(self.operator.sql)
+                log.debug(f"bigquery sql parsed and obtained meta: {sql_meta}")
+
+                return [StepMetadata(
+                    name=get_job_name(task=self.operator),
+                    context={
+                        'sql': self.operator.sql,
+                        'bigquery.sql.parsed.inputs':
+                            json.dumps(sql_meta.in_tables),
+                        'bigquery.sql.parsed.outputs':
+                            json.dumps(sql_meta.out_tables)
+                    }
+                )]
```

Comment:
It doesn't make sense to me that we use the parser only if the other mechanism failed. 
I think the parsing info (`bigquery.sql.parsed...` keys) should be removed or added in all cases so that we can compare.
Also we should add to the context the error from line 73 to facilitate debugging when this doesn't work